lib/transfer.c

```diff
@@ -797,7 +797,7 @@ static CURLcode readwrite_data(struct Curl_easy *data,
                                            nread);
             }
           }
-          else
+          else if(!k->ignorebody)
             result = Curl_unencode_write(conn, k->writer_stack, k->str, nread);
```

Comment:
result is checked right after this in all cases so i'd add this
~~~c
else
  result = CURLE_OK;
~~~
because what if it's set to CURLE_AGAIN and or something at that point, otherwise it's just going to return that and it shouldn't, (unless I'm reading it wrong)

Reply:
It is set to `CURLE_OK` at the top of the function and then it gets checked every time it gets assigned so I don't think there's a risk for that...